traits/observation/_has_traits_helpers.py

```diff
@@ -129,8 +129,7 @@ def ctrait_prevent_event(event):
     """
     if event.old is Uninitialized:
         return True
-
-    ctrait = event.object.traits()[event.name]
+    ctrait = event.object._trait(event.name, 2)
```

Comment:
The `2` here forces instance trait creation, which seems like a potentially surprising side-effect. Is this necessary? If so, should the docstring be updated to indicate the potential side-effect?

Reply:
Would using the `event.object.trait` method be an option here?

Reply:
> The 2 here forces instance trait creation, which seems like a potentially surprising side-effect. Is this necessary?

This function is called when the event has already fired from the CTrait. So persumably the ctrait should have been created already?

Reply:
Anyway, yes I can use `trait` method here.

Reply:
I don't know - I was asking mainly for my understanding. I had the understanding that change handlers are always attached to an instance CTrait, I don't know if that is right.